aptihramy-backend/database.py

```diff
@@ -0,0 +1,180 @@
+import polars as pl
+import blitzbeaver as bb
+from blitzbeaver.literals import ID
+from constants import COLUMN_RAW_TO_PRETTY, COLUMN_PRETTY_TO_RAW
+import time as time
+
+
+class Database:
+
+    def __init__(
+        self,
+        record_schema: bb.RecordSchema,
+        tracking_config: bb.TrackerConfig,
+        path_graph: str,
+        csv_path: str,
+        start_year: int,
+        end_year: int,
+    ):
+        """
+        Initializes the Database instance.
+
+        Args:
+            record_schema (bb.RecordSchema): The schema defining the records.
+            tracking_config (bb.TrackerConfig): The tracking configuration.
+            path_graph (str): Path to the Beaver diagnostic graph file.
+            csv_path (str): Path to the folder containing CSV data used to create the beaver file.
+            start_year (int): The starting year of the dataset.
+            end_year (int): The ending year of the dataset.
+        """
+        self._record_schema = record_schema
+        self._tracking_config = tracking_config
+        self._path_graph = path_graph
+        self._csv_path = csv_path
+        self._start_year = start_year
+        self._end_year = end_year
+        self._graph = bb.read_beaver(path_graph)
+        self._feature_indexes = self._get_feature_indexes()
+        self._feature_last_frame_value = self._get_last_frame_values_for_all_features()
+
+    def _get_feature_indexes(self) -> dict[str, int]:
+        d = {}
+        for i, field in enumerate(self._record_schema.fields):
+            d[field.name] = i
+        return d
+
+    def _get_last_frame_values_for_all_features(self) -> dict[str, dict[ID, list[str]]]:
+        d = {}
+        for schema in self._record_schema.fields:
+            d[schema.name] = self.get_last_frame_values_of_feature(schema.name)
+        return d
+
+    def get_dataframes(self) -> list[pl.DataFrame]:
+        """
+        Reads CSV files for each year in the range and returns them as a list of Polars DataFrames.
+
+        Returns:
+            list[pl.DataFrame]: A list of DataFrames loaded from CSV files.
+        """
+        return [
+            pl.read_csv(f"{self._csv_path}/{year}.csv", infer_schema_length=10000)
+            for year in range(self._start_year, self._end_year + 1)
+        ]
+
+    def get_feature_index(self, raw_feature: str) -> int | None:
+        return self._feature_indexes[raw_feature]
```

Comment:
If the method returns `int | None`, you should use `self._feature_indexes.get(raw_feature)`, `self._feature_indexes[raw_feature]` will raise an exception if the feature doesn't exist